### PR TITLE
Update Chrome and Safari data and add Error.cause test

### PR DESCRIPTION
**Pull request by chicoxyzzy**
Also see comments

### CODE REVIEW

File: data-esnext.js [hunk 1]
```diff
@@ -1431,7 +1434,7 @@ exports.tests = [
     babel7corejs2: true,
     babel7corejs3: true,
     chrome1: false,
-    chrome91: true,
+    chrome92: false,
```
Comment (chicoxyzzy) [inline data-esnext.js]: This one was added mistakenly probably

File: data-es2016plus.js [hunk 2]
```diff
@@ -5742,6 +5742,7 @@ exports.tests = [
       ie11: false,
       opera10_50: false,
       safari12: false,
+      safari15: true,
```
Comment (chicoxyzzy) [inline data-es2016plus.js:5745]: Tested both on macOS Monterey and iOS 15

File: data-esnext.js [hunk 3]
```diff
@@ -1543,6 +1548,26 @@ exports.tests = [
...
   ]
+},
+{
+  name: 'Error.cause property',
+  category: STAGE3,
+  significance: 'small',
+  spec: 'https://github.com/tc39/proposal-error-cause',
+  exec: function () {/*
+      var error = new Error('error', { cause: 'cause' })
+      return error.hasOwnProperty('cause') && error.cause === 'cause';
```
Comment (ljharb) [inline data-esnext.js:1566]: we may want to test all 8 of the error constructors instead of just this 1

**Review by ljharb — APPROVED** [review]

File: data-esnext.js (hunk 1 above)
Comment (nicolo-ribaudo) [reply data-esnext.js]: This code returns `true` for me in Chromium 90 :thinking: 

```js
(() => {
      let ok = false;
    class A {
      static { ok = true; }
    }
    return ok;
 })();
```

Comment (nicolo-ribaudo) [reply data-esnext.js]: Oh I had the "Experimental JavaScript" flag enabled.

Comment (chicoxyzzy) [reply data-esnext.js]: Thanks, I will add it under flag

Comment (chicoxyzzy) [reply data-esnext.js]: @nicolo-ribaudo are you sure it's Chrome 90? I see Chrome 91 mentioned [here](https://v8.dev/features/class-static-initializer-blocks). I have only Chrome 92 installed on this machine, so can't check

File: data-esnext.js (hunk 3 above)
Comment (chicoxyzzy) [reply data-esnext.js:1566]: done

**Review by ljharb — APPROVED** [review]

File: data-esnext.js (hunk 1 above)
Comment (chicoxyzzy) [reply data-esnext.js]: It was 91 in https://github.com/kangax/compat-table/pull/1716, so I'm going to merge current version for now, we can change to 90 later if necessary

Comment (nicolo-ribaudo) [reply data-esnext.js]: I use Brave (which is Chromium-based), and the `chrome://version` page says `Chromium: 90.0.4430.212` and `V8 9.0.257.29`.

Probably it's flagged under Chrome 90 and unflagged in Chrome 91.

Comment (chicoxyzzy) [reply data-esnext.js]: It's not available in Chrome 92 to 94 by default

**Pull request merged**